### PR TITLE
you kids like extend votes?

### PR DESCRIPTION
**Pull request by Hatterhat**
## About The Pull Request
extend votes. stolen from quotefox/Hyper-Station-13#96
## Why It's Good For The Game
green shifts are bad but you can make them exponentially worse by voting on this
## Changelog
:cl:
add: Extend votes! Ported from Hyper, ported from AUstation.
/:cl:

### CODE REVIEW

**Review by silicons — DISMISSED** [review]

**Comment by Ghommie** [comment]
Make it compile.

**Comment by Arturlang** [comment]
Why not port the add antag option from bay for !!FUN!!

File: code/controllers/configuration/entries/general.dm [hunk 1]
```diff
@@ -82,6 +82,16 @@
 	config_entry_value = 600
 	min_val = 0
 
+/datum/config_entry/number/vote_autotransfer_initial //length of time before the first autotransfer vote is called (deciseconds, default 2 hours)
+	config_entry_value = 72000
+	integer = FALSE
```
Comment (Ghommie) [inline code/controllers/configuration/entries/general.dm]: The value is in deciseconds and byond ticks are 1 ds minimum. I don't understand why it should be a float.

File: code/controllers/configuration/entries/general.dm [hunk 2]
```diff
@@ -82,6 +82,16 @@
...
+	min_val = 0
+
+/datum/config_entry/number/vote_autotransfer_interval //length of time to wait before subsequent autotransfer votes (deciseconds, default 30 minutes)
+	config_entry_value = 18000
+	integer = FALSE
```
Comment (Ghommie) [inline code/controllers/configuration/entries/general.dm]: Same.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by keronshb** [comment]
Honestly I do see the good in this but also the bad.

Some rounds are suppose to be short fun things (Wiz, Nukies, Gangs, maybe Cult if they're speedrunning)

Most should be a good medium run (Cult, Traitors, Bloodsuckers, etc), but I feel this might turn those into big shitter safari rounds.  These are the 1:30 - 2:30 hr rounds.

I know this would be left up to the players to vote in game, but rounds end up going past 2:30 (even then it feels too long), people might get round fatigued a lot more.  Especially if people start rallying behind votes like they use to when they weren't hidden for dynamic.

This feels like a big blow also for the tots or cult that was caught too early + killed and then it's "forced extended" for the rest of the shift and if it gets voted... even longer?  I know ghost roles do exist, but just like the station you can only do so much with them in the current allotted time that going past that feels like an eternity.

I feel like if this does stay, midrounds should be 2x or 3x the appearance rate when the round is voted to be extended.

EDIT:

Yes inb4 the 4head response "There's OTHER servers and you can just go do something else!"  Imagine waiting THAT long though?  Coming back 2+ hours later and the round is still going?

**Review by Ghommie — CHANGES_REQUESTED** [review]
Remember to remove the shuttle subsystem's `auto_call` variable and modify its `autoEnd` procedure accordingly.

**Review by Ghommie — APPROVED** [review]

File: code/controllers/subsystem/shuttle.dm [hunk 3]
```diff
@@ -643,10 +643,11 @@ SUBSYSTEM_DEF(shuttle)
...
-	if((world.realtime - SSshuttle.realtimeofstart) > auto_call && EMERGENCY_IDLE_OR_RECALLED) //2 hours
+/datum/controller/subsystem/shuttle/proc/autoEnd() //CIT CHANGE - allows shift to end without being a proper shuttle call?
+	if(EMERGENCY_IDLE_OR_RECALLED)
 		SSshuttle.emergency.request(silent = TRUE)
 		priority_announce("The shift has come to an end and the shuttle called. [seclevel2num(get_security_level()) == SEC_LEVEL_RED ? "Red Alert state confirmed: Dispatching priority shuttle. " : "" ]It will arrive in [emergency.timeLeft(600)] minutes.", null, "shuttlecalled", "Priority")
-		log_game("Round time limit reached. Shuttle has been auto-called.")
-		message_admins("Round time limit reached. Shuttle called.")
+		log_game("Round end vote passed. Shuttle has been auto-called.")
+		message_admins("Round end vote passed. Shuttle has been auto-called.")
 		emergencyNoRecall = TRUE
```
Comment (Ghommie) [inline code/controllers/subsystem/shuttle.dm]:
```suggestion
	emergencyNoRecall = TRUE
```

File: code/controllers/subsystem/shuttle.dm [hunk 4]
```diff
@@ -643,10 +643,11 @@ SUBSYSTEM_DEF(shuttle)
...
+		endvote_passed = TRUE
```
Comment (Ghommie) [inline code/controllers/subsystem/shuttle.dm]:
```suggestion
	endvote_passed = TRUE
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Pull request merged**